### DISCUSSION TITLE
Add first-class support for Pug in .astro files

### DISCUSSION THREAD

**Discussion by KenEucker** [withastro/roadmap#885] [Proposal]
### Body

# Summary

Pug templates are much more concise and quicker to implement compared to all of the requirements of HTML. Adding built-in support for pug templates in astro files would greatly reduce the size of source files.

# Background & Motivation

I took on the task of creating this discussion after reading through [this issue on GitHub](https://github.com/withastro/astro/issues/1513) and realizing that it hadn't been proposed here yet. 

Users have reported the following:

"I've used Pug for years. I prefer it because life is too short to waste it closing opening tags (yes, I'm looking at you, HTML). Beyond the improved authoring experience, however, It also provides a higher signal-to-noise ratio when reading template code, which is more of a benefit than I think a lot of people may realize."

"If Astro offered first-class support for Pug in .astro files it may become the closest-to-perfect SSG in existence (just sayin')."

"It'd be nice if we could simply write Pug in place of Html in the way that .astro files current support Html."

# Goals

- add pug support to astro files
- have .pug files treated similar to markdown files (and exposing Astro.props to them for dynamic stuff) 
- be able to import a pug component from 'astro/components' for placing pug templates into astro templates

# Example

```
---
import { Pug } from 'astro/components';
---

<Pug>
    h1 Hello!
    p this is a test
</Pug>
```

**Comment by louiss0** [comment]
I'm a fan of this idea. I think a pug renderer would also be a good idea.

**Comment by nellePoint** [comment]
this would be wonderful.

**Comment by vai** [comment]
yes please.

**Comment by lordfpx** [comment]
Yes, please, that will help to migrate Pug projects to Astro

**Comment by sergio-agosti** [comment]
Agree, I love the pug syntax and would love to see this happening...
